File: c/01-estrutura-sequencial/idades.c
#include <stdio.h> 
#include <string.h>

void read_text(char * buffer, int lenght) {
    fgets(buffer, lenght, stdin);
    strtok(buffer, "\n");
}

void input_cleanup () {
    char c;
    while ((c = getchar()) != '\n' && c != EOF) { }
}

int main () {
    int age1, age2;
    double media;
    char name1[50], name2[50]; 
    
    printf("Dados da primeira pessoa: \n");
    printf("Nome: ");
    read_text(name1, 50);
    
    printf("Idade: ");
    scanf("%d", &age1);

    printf("Dados da segunda pessoa: \n");
    printf("Nome: ");
    input_cleanup();
    read_text(name2, 50);
    
    printf("Idade: ");
    scanf("%d", &age2);

    media = (age1 + age2) / 2.0;

    printf("A idade media de %s e %s é de %.1lf anos\n", name1, name2, media);
    
    return 0;
}

File: java/01-estrutura-sequencial/Idades.java

import java.util.Locale;
import java.util.Scanner;

public class Idades {

	public static void main(String[] args) {
		
		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);
		
		String name1, name2;
		int age1, age2;
		double average;
		
		System.out.println("Dados da primeira pessoa: ");
		System.out.print("Nome: ");
		name1 = sc.nextLine();
		System.out.print("Idade: ");
		age1 = sc.nextInt();
		
		sc.nextLine();
		
		System.out.println("Dados da segunda pessoa: ");
		System.out.print("Nome: ");
		name2 = sc.nextLine();
		System.out.print("Idade: ");
		age2 = sc.nextInt();
		
		average = (double) (age1 + age2) / 2.0;
		
		System.out.println("A idade media de " + name1 + " e " + name2 + " é de " + String.format("%.1f", average) + " anos");
		
		sc.close();
	}

}
